modules/@apostrophecms/modal/ui/apos/mixins/AposDocsManagerMixin.js

```diff
@@ -146,6 +149,32 @@ export default {
     saveRelationship() {
       this.$emit('modal-result', this.checkedDocs);
       this.modal.showModal = false;
+    },
+    // Default implementation of isModified is based on whether the
+    // selection has changed, but you can override this and combine
+    // that bit with your own if your manager allows in-context editing
+    // of a piece (i.e. AposMediaManager)
+    isModified() {
+      return this.relationshipIsModified();
+    },
+    // Easy to reuse if you have a custom isModified method
+    relationshipIsModified() {
+      if (!this.relationshipField) {
+        return false;
+      }
+      if (this.chosen.length !== this.checkedDocs.length) {
+        return true;
+      }
+      for (let i = 0; (i < this.chosen.length); i++) {
+        if (this.chosen[i]._id !== this.checkedDocs[i]._id) {
+          return true;
+        }
+        if (this.relationshipField.schema) {
```

Comment:
Currently you can't edit the relationship in our manager-as-chooser modals, but this could change, so this is good futureproofing.